interop-testing/src/main/java/io/grpc/testing/integration/CustomBackendMetricsLoadBalancerProvider.java

```diff
@@ -0,0 +1,154 @@
+  private volatile TestOrcaReport latestOobReport;
+
+  @Override
+  public LoadBalancer newLoadBalancer(LoadBalancer.Helper helper) {
+    return new CustomBackendMetricsLoadBalancer(helper);
+  }
+
+  @Override
+  public boolean isAvailable() {
+    return true;
+  }
+
+  @Override
+  public int getPriority() {
+    return 0;
+  }
+
+  @Override
+  public String getPolicyName() {
+    return TEST_ORCA_LB_POLICY_NAME;
+  }
+
+  private final class CustomBackendMetricsLoadBalancer extends ForwardingLoadBalancer {
+    private LoadBalancer delegate;
+
+    public CustomBackendMetricsLoadBalancer(Helper helper) {
+      this.delegate = LoadBalancerRegistry.getDefaultRegistry()
+          .getProvider("pick_first")
+          .newLoadBalancer(new CustomBackendMetricsLoadBalancerHelper(helper));
+    }
+
+    @Override
+    public LoadBalancer delegate() {
+      return delegate;
+    }
+
+    private final class CustomBackendMetricsLoadBalancerHelper
+        extends ForwardingLoadBalancerHelper {
+      private final Helper orcaHelper;
+
+      public CustomBackendMetricsLoadBalancerHelper(Helper helper) {
+        this.orcaHelper = OrcaOobUtil.newOrcaReportingHelper(helper);
+      }
+
+      @Override
+      public Subchannel createSubchannel(CreateSubchannelArgs args) {
+        Subchannel subchannel = super.createSubchannel(args);
+        OrcaOobUtil.setListener(subchannel, new OrcaOobUtil.OrcaOobReportListener() {
+              @Override
+              public void onLoadReport(OrcaLoadReport orcaLoadReport) {
+                latestOobReport = fromOrcaLoadReport(orcaLoadReport);
+              }
+            },
+            OrcaOobUtil.OrcaReportingConfig.newBuilder()
+                .setReportInterval(1, TimeUnit.SECONDS)
+                .build()
+        );
+        return subchannel;
+      }
+
+      @Override
+      public void updateBalancingState(ConnectivityState newState, SubchannelPicker newPicker) {
+        delegate().updateBalancingState(newState, new MayReportLoadPicker(newPicker));
+      }
+
+      @Override
+      public Helper delegate() {
+        return orcaHelper;
+      }
+    }
+
+    private final class MayReportLoadPicker extends SubchannelPicker {
+      private SubchannelPicker delegate;
+
+      public MayReportLoadPicker(SubchannelPicker delegate) {
+        this.delegate = delegate;
+      }
+
+      @Override
+      public PickResult pickSubchannel(PickSubchannelArgs args) {
+        PickResult result = delegate.pickSubchannel(args);
+        if (result.getSubchannel() == null) {
+          return result;
+        }
+        AtomicReference<TestOrcaReport> reportRef =
+            args.getCallOptions().getOption(ORCA_OOB_REPORT_KEY);
+        if (reportRef != null) {
+          reportRef.set(latestOobReport);
+        }
+
+        return PickResult.withSubchannel(
+            result.getSubchannel(),
+            OrcaPerRequestUtil.getInstance().newOrcaClientStreamTracerFactory(
```

Comment:
Hmmm... Ideally we'd pass `result.getStreamTracerFactory()` as the first argument to `newOrcaClientStreamTracerFactory()`. But it looks like `newOrcaClientStreamTracerFactory()` requires the argument to be non-null, which makes this pretty annoying to use from a non-leaf policy. Let's change `newOrcaClientStreamTracerFactory()` to allow null and then swap to the two-arg version here?

No need to do that in this PR. It can be a follow-up.